src/editor.ts

```diff
@@ -272,14 +272,21 @@ export class PreviewProvider implements WebviewViewProvider {
       localResourceRoots: [ctx.ext.extensionUri],
     }
 
-    // TODO: get port from process info
-    const serverAddr = 'http://localhost:3030/'
+    const serverAddr = `http://localhost:${getServerPort()}/`
     const url = `${serverAddr}${idx}?embedded=true`
-
     try {
       await got.get(`${serverAddr}index.html`, { responseType: 'text', resolveBodyOnly: true })
     }
     catch {
+      window.showErrorMessage(`The ${serverAddr} page isn't work, please run slide dev or set the correct port of the server`, 'set the port').then(async command => {
```

Comment:
The webview already shows a message when server not found. Better to have a button on the view which bind to a command, instead of prompting (could be annoying)